dart-parser/src/parser/type_params.rs

```diff
@@ -0,0 +1,49 @@
+use nom::{
+    bytes::complete::tag,
+    combinator::{cut, opt},
+    error::{context, ContextError, ParseError},
+    multi::separated_list1,
+    sequence::{pair, preceded, terminated, tuple},
+    Parser,
+};
+
+use crate::dart::TypeParam;
+
+use super::{
+    common::spbr,
+    identifier::{identifier, identifier_ext},
+    PResult,
+};
+
+pub fn type_params<'s, E>(s: &'s str) -> PResult<Vec<TypeParam>, E>
```

Comment:
@Taym95 Sorry for taking over. I hope you didn't spend too much effort on your implementation.